rewrite-maven/src/main/java/org/openrewrite/maven/AddDependency.java

```diff
@@ -124,6 +125,40 @@ public class AddDependency extends Recipe {
     @With
     String familyPattern;
 
+    @Option(displayName = "Accept transitive",
+            description = "Default false. If enabled, the dependency will not be added if it is already on the classpath as a transitive dependency.",
+            example = "true",
+            required = false)
+    @Nullable
+    Boolean acceptTransitive;
+
+    @JsonCreator
+    public AddDependency(String groupId, String artifactId, String version, @Nullable String versionPattern,
+            @Nullable String scope, @Nullable Boolean releasesOnly, String onlyIfUsing, @Nullable String type,
+            @Nullable String classifier, @Nullable Boolean optional, @Nullable String familyPattern,
+            @Nullable Boolean acceptTransitive) {
```

Comment:
Let's not add multiple constructors. It has caused us problems in the past. It is fine to add a new field without adding a new constructor for backwards compatibility. If there are a lot of recipes that use this recipe directly in code, rather than through yaml, we could use lombok's `@Builder` annotation

Reply:
Sure, I can make that change - I assumed preserving the backwards compatibility would be less disruptive overall, but good to know the preference.

Out of curiosity, what sort of issues? Usual hygiene stuff (eg removing the deprecation is just another task for later and some references will inevitably still break anyway when you do) or maybe some Lombok nuance?

Reply:
done @sambsnyd